### PR TITLE
Branch table feature

### PR DESCRIPTION
**Pull request by LINCHENYU2030S**
add table command

### CODE REVIEW

**Review by ChuanXinNg — CHANGES_REQUESTED** [review]
LGTM. Can be merged after some modification. I don't really get why is type long used for the counts, I think int is enough to hold the counts🤔.

File: src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import seedu.address.commons.util.ToStringBuilder;
+
+
+/**
+ * Represents the command result for gender table generation.
+ */
+public class GenderTableCommandResult extends CommandResult {
+    private static final String feedbackToUserMessage = "A table categorized by gender is shown";
+    private Long maleCount;
+    private Long femaleCount;
```
[the remark this answers is not in the document]
Comment (LINCHENYU2030S) [reply src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java:15]: This is actually the return type of java stream count() method and the Long data type for table count is only used in table-related stuff.

File: src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+
+/**
+ * Represents the command result for gender table generation.
+ */
+public class GenderTableCommandResult extends CommandResult {
+    private static final String feedbackToUserMessage = "A table categorized by gender is shown";
```
[the remark this answers is not in the document]
Comment (LINCHENYU2030S) [reply src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java:13]: Thanks for pointing it out!

File: src/main/java/seedu/address/logic/commands/TableCommand.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+     */
+    public TableCommand(String args) {
+        this.args = args.trim();
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        switch(this.args) {
+        case "g/":
```
[the remark this answers is not in the document]
Comment (LINCHENYU2030S) [reply src/main/java/seedu/address/logic/commands/TableCommand.java:55]: Thanks for pointing out! I just notice prefix has instance method getPrefix() which will return me the corresponding prefix in String.

File: src/main/java/seedu/address/logic/commands/TableCommand.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    /**
+     * Generate SecLevelTableCommandResult instance
+     * @param model instance of Model subclass, e.g. ModelManager instance
+     * @return SecLevelTableCommandResult instance containing column titles and values.
+     */
+    private SecLevelTableCommandResult executeSecLevel(Model model) {
+        Map<String, Long> columnValueMapping = new HashMap<>();
+        String[] titles = new String[] {"Sec 1", "Sec 2", "Sec 3", "Sec 4"};
```
[the remark this answers is not in the document]
Comment (LINCHENYU2030S) [reply src/main/java/seedu/address/logic/commands/TableCommand.java]: Noted with thanks!

File: src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (LINCHENYU2030S) [reply src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java:15]: I think we may leave it as Long first, and we can change it if it is not good to fit in later implementation.

[the remark this answers is not in the document]
Comment (LINCHENYU2030S) [reply src/main/java/seedu/address/logic/commands/GenderTableCommandResult.java:15]: just find out that java steam has mapToInt method which I can used to convert Long to Int in just one to two lines, will update now

**Review by ChuanXinNg — APPROVED** [review]
LGTM. Modifications on the previous comments are made and good job for adding use cases.

**Pull request merged**